File: backend/src/main/java/edu/cit/cabasag/barberconnect/model/HaircutStyle.java
package edu.cit.cabasag.barberconnect.model;

import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class HaircutStyle {
    
    private String haircut_style_id;
    private String barber_profile_id; // Reference to BarberProfile barber_profile_id
    private String name;
    private String description;
    private BigDecimal basePrice;
    private Integer durationMinutes;
    private String imageUrl;
    private Boolean isActive = true;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    
    // References to related data
    private List<String> styleOptionIds; // References to StyleOption style_option_ids
}

File: mobile/app/src/main/java/edu/cit/cabasag/barberconnect/model/HaircutStyle.kt
package edu.cit.cabasag.barberconnect.model

import com.google.gson.annotations.SerializedName

data class HaircutStyle(
    @SerializedName("haircut_style_id") val haircutStyleId: String? = null,
    @SerializedName("barber_profile_id") val barberProfileId: String? = null,
    @SerializedName(value = "name", alternate = ["styleName", "serviceName"]) val name: String? = null,
    val description: String? = null,
    @SerializedName(value = "basePrice", alternate = ["base_price", "price"]) val basePrice: Double? = null,
    @SerializedName(value = "durationMinutes", alternate = ["duration_minutes", "duration"]) val durationMinutes: Int? = null,
    @SerializedName(value = "imageUrl", alternate = ["image_url"]) val imageUrl: String? = null,
    @SerializedName(value = "isActive", alternate = ["is_active"]) val isActive: Boolean? = null
)
